plugins/outputs/socket_writer/socket_writer.go

```diff
@@ -55,13 +58,39 @@ func (sw *SocketWriter) Connect() error {
 	}
 
 	var c net.Conn
-	if tlsCfg == nil {
-		c, err = net.Dial(spl[0], spl[1])
+
+	if spl[0] == "vsock" {
+		addrTuple := strings.SplitN(spl[1], ":", 2)
+
+		//check if address string has two tokens
```

Comment:
See comment on input
```suggestion
		// Check address string for containing two tokens
```